### ISSUE TITLE
Compatibility Problem with javafx-8.0.jar

### ISSUE THREAD

**Issue by elkassar** [TomasMikula/ReactFX#60]
`Val<T>` extends `javafx.beans.value.ObservableValue`. However, **"javafx-8.0.jar"** is using another version  of `javafx.beans.value.ObservableValue`. So, There is a conflict happens whenever we use both jars in the same project.

**Comment by TomasMikula** [comment]
Hi, what is "javafx-8.0.jar" and where does it come from?

**Comment by elkassar** [comment]
It is required in my project to load the system library "jfxrt.jar" classes into my project.

groupId: com.oracle
artifactId: javafx
version: 8.0

**Comment by TomasMikula** [comment]
I still don't understand, does that jar define its own `javafx.beans.value.ObservableValue`? Does the error arise because you are trying to load a class with the same name twice? What is the error message you get? What JDK version do you use?

**Comment by elkassar** [comment]
Yes, this jar defines its own `javafx.beans.value.ObservableValue`, and may be it is a wrapper of this class which is inside "jfxrt.jar" which is existing inside the JDK. Yes the problem arise because there are two different instances of this class inside both jars. I'm using jdk1.8.0_91 or jdk1.8.0_73. Both JDKs have the same problem. The error message is 

> "java.lang.LinkageError: loader constraint violation: when resolving method "MyClass.<init>(LMyClass;Ljavafx/beans/value/ObservableValue;Z)V" the class loader (instance of org/eclipse/osgi/internal/loader/EquinoxClassLoader) of the current class, MyClass2, and the class loader (instance of org/eclipse/osgi/internal/loader/EquinoxClassLoader) for the method's defining class, MyClass3, have different Class objects for the type javafx/beans/value/ObservableValue used in the signature"

.

Whenever I remove the dependency of any of "**javafx-8.0.jar**" or "**reactfx-2.0-M5.jar**", the error disappear.

**Comment by TomasMikula** [comment]
Well, ReactFX doesn't define `javafx.beans.value.ObservableValue`, it just uses the one from JDK. I don't understand why you would need an extra jar file (`javafx-8.0.jar`) when JavaFX is part of the JDK already. You can try to compile ReactFX yourself with a dependency on that jar, to see if that resolves the problem.
